services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt

```diff
@@ -72,563 +72,597 @@ class HankeControllerITests(@Autowired override val mockMvc: MockMvc) : Controll
+                disclosureLogService.saveDisclosureLogsForHanke(hanke, USERNAME)
+            }
+        }
 
-    @Test
-    fun `Add Hanke and return it newly created hankeTunnus (POST)`() {
-        val hankeToBeMocked =
-            HankeFactory.create(
-                id = null,
-                hankeTunnus = null,
-                version = null,
-                createdAt = null,
-                createdBy = null,
-            )
-        val createdHanke =
-            hankeToBeMocked.copy(
-                id = 1,
-                hankeTunnus = HANKE_TUNNUS,
-                version = 0,
-                createdBy = USERNAME,
-                createdAt = getCurrentTimeUTC(),
-            )
-        every { hankeService.createHanke(any()) }.returns(createdHanke)
-        justRun { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
-
-        postRaw(BASE_URL, hankeToBeMocked.toJsonString())
-            .andExpect(status().isOk)
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
-            .andExpect(jsonPath("$.hankeTunnus").value(HANKE_TUNNUS))
-            .andExpect(content().json(createdHanke.toJsonString()))
-
-        verify { hankeService.createHanke(any()) }
-        verify { disclosureLogService.saveDisclosureLogsForHanke(createdHanke, USERNAME) }
-        verify { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
+        @Test
+        fun `With known hanketunnus return applications`() {
+            val hanke = HankeFactory.create()
+            val applications = AlluDataFactory.createApplications(5)
+            every { hankeService.getHankeWithApplications(HANKE_TUNNUS) } returns
+                HankeWithApplications(hanke, applications)
+            every {
+                permissionService.hasPermission(hanke.id!!, USERNAME, PermissionCode.VIEW)
+            } returns true
+
+            val response: ApplicationsResponse = get(url).andExpect(status().isOk).andReturnBody()
+
+            assertTrue(response.applications.isNotEmpty())
+            assertEquals(ApplicationsResponse(applications), response)
+            verifySequence {
+                hankeService.getHankeWithApplications(HANKE_TUNNUS)
+                permissionService.hasPermission(hanke.id!!, USERNAME, PermissionCode.VIEW)
+                disclosureLogService.saveDisclosureLogsForHanke(hanke, USERNAME)
+                disclosureLogService.saveDisclosureLogsForApplications(applications, USERNAME)
+            }
+        }
     }
 
-    @Test
-    fun `createHanke when perustaja is provided should return provided information`() {
-        val hanke = HankeFactory.create().withPerustaja()
-        every { hankeService.createHanke(any()) } returns hanke
-        justRun { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
+    @Nested
+    inner class CreateHanke {
+        private val url = BASE_URL
```

Comment:
Is this necessary?

Reply:
Strictly speaking, it's not necessary here, `BASE_URL` could be used as-is. Still, I added it to keep the pattern with the other nested classes.

The reason I added the `url` variables was that in some test returning 404, the URL was incorrect, giving the correct HTTP response, but because of wrong reasons. Having the same URL in all tests of the same endpoint makes sure this can't happen.